network/validator/pubsub/staked_validator.go

```diff
@@ -0,0 +1,20 @@
+package validator
+
+import (
+	"context"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/network/message"
+)
+
+func StakedValidator(getIdentity func(peer.ID) (*flow.Identity, bool)) MessageValidator {
+	return func(ctx context.Context, from peer.ID, msg *message.Message) pubsub.ValidationResult {
```

Comment:
This would probably need an adjustment, as we would want to check that the origin of the message (rather than the last relayer, which is the `from` argument here) is `getIdentity`-valid.

See my comment below on the inner call to validators in `TopicValidator`.

It's not too cumbersome a fix, since a call to `messageSigningID` does not require / involve deserialization.

Reply:
See comment above. Everything here is how I intended it to be.